RR.App.Tests/Controllers/HRIS/ClientProjectsControllerTests.cs

```diff
@@ -0,0 +1,157 @@
+            Assert.NotNull(okResult);
+            Assert.Equal(expectedClientProjects, okResult.Value);
+        }
+
+        [Fact]
+        public async Task GetClientProject_WithValidId_ReturnsOkResult()
+        {
+            var id = 1;
+            var clientProjectDto = new ClientProjectsDto { Id = id };
+            _mockClientProjectService.Setup(service => service.GetClientProject(id))
+                .ReturnsAsync(clientProjectDto);
+
+            var result = await _controller.GetClientProject(id);
+            var okResult = Assert.IsType<ActionResult<ClientProjectsDto>>(result);
+            Assert.NotNull(okResult);
+            var objectResult = Assert.IsType<OkObjectResult>(okResult.Result);
+            Assert.NotNull(objectResult);
+            Assert.Equal(clientProjectDto, objectResult.Value);
+        }
+
+        [Fact]
+        public async Task DeleteClientProject_WithExistingId_ReturnsNoContent()
+        {
+            var id = 1;
+            _mockClientProjectService.Setup(service => service.GetClientProject(id))
+                .ReturnsAsync(new ClientProjectsDto { Id = id });
+
+            var result = await _controller.DeleteClientProject(id);
+            Assert.IsType<OkObjectResult>(result);
+        }
+
+        [Fact]
+        public async Task PutClientProject_WithMatchingId_ReturnsNoContent()
+        {
+            var id = 1;
+            var clientProject = new ClientProject { Id = id };
+            _mockClientProjectService.Setup(service => service.UpdateClientProject(clientProject))
+                .ReturnsAsync(new ClientProjectsDto());
+
+            var result = await _controller.PutClientProject(id, clientProject);
+            Assert.IsType<OkObjectResult>(result);
+        }
+
+        [Fact]
+        public async Task PostClientProject_WithValidInput_ReturnsCreatedAtAction()
+        {
+            var clientProject = new ClientProject
+            {
+                Id = 1,
+            };
+            var createdClientProjectDto = new ClientProjectsDto { Id = clientProject.Id };
+            _mockClientProjectService.Setup(service => service.CreateClientProject(clientProject))
+                .ReturnsAsync(createdClientProjectDto);
+
+            var result = await _controller.PostClientProject(clientProject);
+            var createdAtActionResult = Assert.IsType<CreatedAtActionResult>(result);
+            Assert.Equal(nameof(_controller.GetClientProject), createdAtActionResult.ActionName);
+            Assert.Equal(clientProject.Id, createdAtActionResult.RouteValues["id"]);
+            Assert.Equal(createdClientProjectDto, createdAtActionResult.Value);
+        }
+
+        [Fact]
+        public async Task PutClientProject_WithNonMatchingId_ReturnsBadRequest()
+        {
+            var clientProject = new ClientProject { Id = 2 };
+            var result = await _controller.PutClientProject(1, clientProject);
+            Assert.IsType<BadRequestObjectResult>(result);
```

Comment:
How do we know if this returns the right object? You need to Assert.Equals to check if it has returned an exprected object